File: app/src/am_viennv/com/example/asian/ImageDiffCallback.java
package com.example.asian;

import androidx.recyclerview.widget.DiffUtil;

import java.util.List;

public class ImageDiffCallback extends DiffUtil.Callback {

    private final List<Image> mOldImages;
    private final List<Image> mNewImages;

    public ImageDiffCallback(List<Image> oldImages, List<Image> newImages) {
        this.mOldImages = oldImages;
        this.mNewImages = newImages;
    }

    @Override
    public int getOldListSize() {
        return mOldImages.size();
    }

    @Override
    public int getNewListSize() {
        return mNewImages.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return mOldImages.get(oldItemPosition).getmIdImage().equals(mNewImages.get(newItemPosition).getmIdImage());
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        Image oldImage = mOldImages.get(oldItemPosition);
        Image newImage = mNewImages.get(newItemPosition);
        return oldImage.equals(newImage);
    }
}

File: app/src/main/java/com/example/ex02/view/adapter/ImageDiffCallback.kt
package com.example.ex02.view.adapter

import androidx.recyclerview.widget.DiffUtil
import com.example.ex02.bean.ItemImage

class ImageDiffCallback : DiffUtil.ItemCallback<ItemImage>() {
    override fun areItemsTheSame(oldItem: ItemImage, newItem: ItemImage): Boolean {
       return oldItem.imageUrl == newItem.imageUrl
    }

    override fun areContentsTheSame(oldItem: ItemImage, newItem: ItemImage): Boolean {
        return oldItem == newItem
    }
}